docs(mod-guide): clarify how to deal with solutions (#48237)
freeCodeCamp/freeCodeCamp: docs/moderator-handbook.md

### Dealing with Posted Solutions

> When a user replies in a help thread specific to the freeCodeCamp curriculum with a solution, remove it and either, kindly let them know that solutions are not permitted in help threads in your own words or, use the **Solution Instead of Help** "Canned reply".

Reviewer: ```suggestion
When a user replies in a help thread specific to the freeCodeCamp curriculum with a solution, remove it and kindly let them know that solutions are not permitted in help threads in your own words or, use the **Solution Instead of Help** "Canned reply".
```

Reviewer: Before I resolve this, you are talking about this text? 

> When a user replies in a help thread specific to the freeCodeCamp curriculum with a solution, remove it and kindly let them know that solutions are not permitted in help threads in your own words or, use the **Solution Instead of Help** "Canned reply".

Author: Yes correct. Before when the sentence had the word "and either," it was giving an option between a response that is made up vs a canned one. But now the option has shifted to become between removing and using a made up response OR using a canned response (no removal needed).

Reviewer: make sense 👍

> When a user replies in a help thread specific to the freeCodeCamp curriculum with a solution, remove it and either, kindly let them know that solutions are not permitted in help threads in your own words or, use the **Solution Instead of Help** "Canned reply".

Reviewer 2: ```suggestion
If a user replies in a help thread for the freeCodeCamp curriculum with a solution, remove it and use the **Solution Instead of Help** canned reply (or a similar response in your own words).
```

> If the OP (Original Poster) replies within a curriculum help thread with their final solution, blur it and use the **Blurred Spoiler Solution** "Canned reply".

Reviewer 2: ```suggestion
If the OP (Original Poster) replies within a freeCodeCamp curriculum help thread with their final solution, blur it and use the **Blurred Spoiler Solution** canned reply.
```

> If a user creates a topic asking for feedback on a solution, that topic should be moved to the [feedback subforum] if it is not already there and the solution should be blurred. But if the user posts the solution to show if off, unlist the topic.

Reviewer 2: ```suggestion
If a user creates a thread asking for feedback on a solution, move the thread to the feedback subforum and the blur the solution, as necessary. If the user is only posting the solution to show if off, then unlist the thread and use the **Solutions Thread** canned reply.
```

Author: There is a small mistake in the suggestion here "and the blur".

I am also not sure of the meaning of "as necessary". Do we need that phrase?

Reviewer 2: 'As necessary' or 'if needed' - it might be in the right sub forum and already blurred

Reviewer 3: "blur the solution" -> "spoiler the solution"?

Reviewer 2: The button on the forum says 'blur spoiler' for me and that's what the canned reply says, that that's why I picked that
